File: testcases/x86_testcases/x86_output/test5/test5.c
#include <stdio.h>
#include <stdbool.h>

int add(int x, int y) {
    int ans = x + y;
    return ans;
}

int subtract(int x, int y) {
    int ans = x - y;
    return ans;
}

int multiply(int x, int y) {
    int ans = x * y;
    return ans;
}

int divide(int x, int y) {
    int ans = 0;
    if (y != 0) {
        ans = x / y;
    } else {
        printf("Error: Cannot divide by zero.");
        printf("\n");
    }
    return ans;
}

bool relational(int x, int y) {
    bool ans = (x == y);
    return ans;
}

int assignment(int x, int y) {
    x |= y;
    int ans = x;
    return ans;
}

void main2() {
    int result_add = add(10, 5);
    printf("Result of Addition: ");
    printf("%d", result_add);
    printf("\n");

    int result_subtract = subtract(10, 5);
    printf("Result of Subtraction: ");
    printf("%d", result_subtract);
    printf("\n");

    int result_multiply = multiply(10, 5);
    printf("Result of Multiplication: ");
    printf("%d", result_multiply);
    printf("\n");

    int result_divide = divide(10, 5);
    printf("Result of Division: ");
    printf("%d", result_divide);
    printf("\n");

    bool result_relational = relational(10, 5);
    printf("Result of Relational Operation: ");
    printf("%d", result_relational);
    printf("\n");

    int result_assignment = assignment(10, 5);
    printf("Result of Assignment Operation: ");
    printf("%d", result_assignment);
    printf("\n");

    return;
}

int main() {
    main2();
    return 0;
}

File: testcases/x86_testcases/test5.py
def add(x: int, y: int) -> int:
    ans : int = x + y
    return ans

def subtract(x: int, y: int) -> int:
    ans : int = x - y
    return ans

def multiply(x: int, y: int) -> int:
    ans : int = x * y
    return ans

def divide(x: int, y: int) -> int:
    ans : int = 0
    if y != 0:
        ans = x / y
    else:
        print("Error: Cannot divide by zero.")
    return ans

def relational(x : int, y : int) -> bool:
    ans : bool = ( x == y)
    return ans

def assignment(x : int, y : int) -> int:
    x |= y
    ans : int = x
    return ans

def main():
    result_add : int = add(10, 5)
    print("Result of Addition: ",result_add)

    result_subtract : int = subtract(10, 5)
    print("Result of Subtraction: ", result_subtract)

    result_multiply : int = multiply(10, 5)
    print("Result of Multiplication: ", result_multiply)

    result_divide : int = divide(10, 5)
    print("Result of Division: ", result_divide)

    result_relational : int = relational(10, 5)
    print("Result of Relational Operation: ", result_relational)

    result_assignment : int = assignment(10, 5)
    print("Result of Assignment Operation: ", result_assignment)

if __name__ == "__main__":
    main()
